src/Sarif.WorkItems/SarifWorkItemContext.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public class SarifWorkItemContext : PropertiesDictionary
+    {
+        public SarifWorkItemContext() { }
+
+        public SarifWorkItemContext(SarifWorkItemContext initializer) : base(initializer) { }
+
+        internal void InitializeFromLog(SarifLog sarifLog)
+        {
+            // TODOD: Initialize source and non-source-controlled files here,
+            //        as determined by scraping all URLs associated with results.
+            // 
+            //        https://github.com/microsoft/sarif-sdk/issues/1752
+        }
+        public Uri ProjectUri
+        {
+            get { return this.GetProperty(ProjectUriOption); }
+            set { this.SetProperty(ProjectUriOption, value); }
+        }
+
+        public string SecurityToken 
```

Comment:
>SecurityToken [](start = 22, length = 13)

Are we going to persist a security token to a SARIF log file? That sounds scary.

Reply:
Not a Sarif log file, the configuration file.  I agree it might be a good idea not to persist the value on this one, though.

---
In reply to: [381009882](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381009882) [](ancestors = 381009882)

Reply:
This goes to a configuration file. A configuration file on a developer box should ok if the file is in a properly acl'ed location (and if the box is properly locked down to an appropriate set of users). We could do something like insist that these files originate in a disk location we know is ACL'ed, like app user, at the cost of making this file maintenance a bit more inconvenient. This is what the new ASP.NET developer secrets files feature does. Let's talk more offline on this.

---
In reply to: [381010592](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381010592) [](ancestors = 381010592,381009882)

Reply:
I recently saw a project where what they put into the config file was a string like `KeyVault:SecurityTokenSecret`, and their config file handling code recognizes the prefix and resolves the value from the key vault.

---
In reply to: [381634480](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381634480) [](ancestors = 381634480,381010592,381009882)